src/query/graphite/storage/m3_wrapper.go

```diff
@@ -120,6 +121,24 @@ func translateQuery(query string, opts FetchOptions) (*storage.FetchQuery, error
 	}, nil
 }
 
+func truncateBoundsToResolution(
+	start time.Time,
+	end time.Time,
+	resolution time.Duration,
+) (time.Time, time.Time) {
+	truncatedStart := start.Truncate(resolution)
+	// NB: if truncated start matches start, it's already valid.
+	if truncatedStart.Before(start) {
+		start = truncatedStart.Add(resolution)
+	}
```

Comment:
Hm, shouldn't there be an else here?  Otherwise it seems like `start` doesn't get modified otherwise?  End of this method looks like it `return start, end`.

i.e.
```
	truncatedStart := start.Truncate(resolution)
	// NB: if truncated start matches start, it's already valid.
	if truncatedStart.Before(start) {
		start = truncatedStart.Add(resolution)
	} else {
		start = truncatedStart
	}
```